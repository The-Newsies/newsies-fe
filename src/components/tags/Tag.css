.button {
  display: flex;
  align-items: center;
  justify-content: space-evenly;
  width: 10px;
  background: white;
}

.button:hover {
  cursor: pointer;
}

/* .buttonSelected {
  display: flex;
  align-items: center;
  justify-content: space-evenly;
  background: grey;
  color: black;
  width: 175px;
}

.button p {
  font-size: .8rem;
} */