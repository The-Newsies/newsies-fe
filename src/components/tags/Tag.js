import React from 'react';
import PropTypes from 'prop-types';

export default function Tag({ name, color }) {
  return (
    <div>
      <button>
        <div style={{ background: color }}></div>
        <p>{name}</p>
      </button>
    </div>
  );
}

Tag.propTypes = {
  name: PropTypes.string.isRequired,
  color: PropTypes.string.isRequired
};

