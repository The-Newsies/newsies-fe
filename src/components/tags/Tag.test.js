import React from 'react';
import { shallow } from 'enzyme';
import Tag from './Tag';

describe('Tag component', () => {
  it('renders Tag', () => {
    const wrapper = shallow(<Tag 
      name={'Science'} 
      color={'FFF'}
      fetchByCategory={() => {}}
      setSelectedTag={() => {}}
      isSelected={false}
    />);
    expect(wrapper).toMatchSnapshot();
  });
});


