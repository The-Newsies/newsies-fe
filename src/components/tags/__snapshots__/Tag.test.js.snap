// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tag component renders Tag 1`] = `
<div>
  <button
    onClick={[Function]}
    style={
      Object {
        "alignItems": "center",
        "background": "white",
        "border": "none",
        "display": "flex",
        "justifyContent": "space-evenly",
        "lineHeigth": "18px",
        "padding": "0 5px",
        "textTransform": "uppercase",
        "width": "140px",
      }
    }
  >
    <p>
      Science
    </p>
  </button>
</div>
`;
